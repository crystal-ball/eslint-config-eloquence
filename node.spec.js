const nodeConfigs = require('./node')

describe('Node configs', () => {
  test('The configs match the snapshot', () => {
    expect(nodeConfigs).toMatchSnapshot()
  })
})
