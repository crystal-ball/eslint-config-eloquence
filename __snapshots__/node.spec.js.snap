// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node configs The configs match the snapshot 1`] = `
Object {
  "env": Object {
    "es6": true,
    "node": true,
  },
  "extends": Array [
    "prettier",
  ],
  "overrides": Array [
    Object {
      "env": Object {
        "jest": true,
      },
      "files": Array [
        "*.spec.js",
      ],
      "parserOptions": Object {
        "sourceType": "module",
      },
      "rules": Object {
        "import/first": "off",
      },
    },
  ],
  "parser": "babel-eslint",
  "parserOptions": Object {
    "ecmaVersion": 11,
    "sourceType": "script",
  },
  "plugins": Array [
    "import",
    "prettier",
  ],
  "reportUnusedDisableDirectives": true,
  "rules": Object {
    "accessor-pairs": "error",
    "array-callback-return": Array [
      "error",
      Object {
        "allowImplicit": true,
      },
    ],
    "block-scoped-var": "error",
    "callback-return": "off",
    "camelcase": Array [
      "error",
      Object {
        "properties": "never",
      },
    ],
    "capitalized-comments": Array [
      "off",
      "never",
      Object {
        "block": Object {
          "ignoreConsecutiveComments": true,
          "ignoreInlineComments": true,
          "ignorePattern": ".*",
        },
        "line": Object {
          "ignoreConsecutiveComments": true,
          "ignoreInlineComments": true,
          "ignorePattern": ".*",
        },
      },
    ],
    "class-methods-use-this": Array [
      "error",
      Object {
        "exceptMethods": Array [],
      },
    ],
    "complexity": Array [
      "off",
      11,
    ],
    "consistent-return": "error",
    "consistent-this": "off",
    "constructor-super": "error",
    "default-case": Array [
      "error",
      Object {
        "commentPattern": "^no default$",
      },
    ],
    "default-param-last": "error",
    "dot-notation": Array [
      "error",
      Object {
        "allowKeywords": true,
      },
    ],
    "eqeqeq": Array [
      "error",
      "always",
      Object {
        "null": "ignore",
      },
    ],
    "for-direction": "error",
    "func-name-matching": Array [
      "off",
      "always",
      Object {
        "includeCommonJSModuleExports": false,
      },
    ],
    "func-names": "warn",
    "func-style": Array [
      "off",
      "expression",
    ],
    "getter-return": Array [
      "error",
      Object {
        "allowImplicit": true,
      },
    ],
    "global-require": "off",
    "guard-for-in": "error",
    "handle-callback-err": "off",
    "id-blacklist": "off",
    "id-length": "off",
    "id-match": "off",
    "import/default": "off",
    "import/dynamic-import-chunkname": Array [
      "error",
      Object {
        "importFunctions": Array [],
        "webpackChunknameFormat": "[a-zA-Z-_/.]+",
      },
    ],
    "import/export": "error",
    "import/exports-last": "off",
    "import/extensions": Array [
      "error",
      "ignorePackages",
      Object {
        "js": "never",
        "jsx": "never",
        "mjs": "never",
        "ts": "never",
        "tsx": "never",
      },
    ],
    "import/first": "error",
    "import/group-exports": "off",
    "import/max-dependencies": "off",
    "import/named": "error",
    "import/namespace": "off",
    "import/newline-after-import": "error",
    "import/no-absolute-path": "error",
    "import/no-amd": "error",
    "import/no-anonymous-default-export": Array [
      "error",
      Object {
        "allowAnonymousClass": false,
        "allowAnonymousFunction": false,
        "allowArray": true,
        "allowArrowFunction": false,
        "allowLiteral": true,
        "allowObject": true,
      },
    ],
    "import/no-commonjs": "off",
    "import/no-cycle": Array [
      "error",
      Object {
        "maxDepth": Infinity,
      },
    ],
    "import/no-default-export": "off",
    "import/no-deprecated": "off",
    "import/no-duplicates": "error",
    "import/no-dynamic-require": "error",
    "import/no-extraneous-dependencies": Array [
      "error",
      Object {
        "devDependencies": Array [
          "**/*.{spec,stories}.js",
          "babel.config.js",
          "webpack.config.js",
          "{.jest,.storybook,cypress,test}/**/*",
        ],
      },
    ],
    "import/no-internal-modules": "off",
    "import/no-mutable-exports": "error",
    "import/no-named-as-default": "error",
    "import/no-named-as-default-member": "error",
    "import/no-named-default": "error",
    "import/no-named-export": "off",
    "import/no-namespace": "error",
    "import/no-nodejs-modules": "off",
    "import/no-relative-parent-imports": "off",
    "import/no-restricted-paths": "off",
    "import/no-self-import": "error",
    "import/no-unassigned-import": "off",
    "import/no-unresolved": Array [
      "error",
      Object {
        "caseSensitive": true,
        "commonjs": true,
      },
    ],
    "import/no-unused-modules": "off",
    "import/no-useless-path-segments": "error",
    "import/no-webpack-loader-syntax": "error",
    "import/order": Array [
      "error",
      Object {
        "groups": Array [
          Array [
            "builtin",
            "external",
          ],
          "unknown",
          Array [
            "internal",
            "parent",
            "sibling",
            "index",
          ],
        ],
        "newlines-between": "ignore",
      },
    ],
    "import/prefer-default-export": "off",
    "import/unambiguous": "off",
    "init-declarations": "off",
    "line-comment-position": Array [
      "off",
      Object {
        "applyDefaultPatterns": true,
        "ignorePattern": "",
        "position": "above",
      },
    ],
    "lines-around-directive": Array [
      "error",
      Object {
        "after": "always",
        "before": "always",
      },
    ],
    "lines-between-class-members": Array [
      "error",
      "always",
      Object {
        "exceptAfterSingleLine": false,
      },
    ],
    "max-classes-per-file": Array [
      "off",
      1,
    ],
    "max-depth": Array [
      "off",
      4,
    ],
    "max-lines": Array [
      "off",
      Object {
        "max": 300,
        "skipBlankLines": true,
        "skipComments": true,
      },
    ],
    "max-lines-per-function": Array [
      "off",
      Object {
        "IIFEs": true,
        "max": 50,
        "skipBlankLines": true,
        "skipComments": true,
      },
    ],
    "max-nested-callbacks": "off",
    "max-params": Array [
      "error",
      3,
    ],
    "max-statements": Array [
      "off",
      10,
    ],
    "max-statements-per-line": Array [
      "off",
      Object {
        "max": 1,
      },
    ],
    "multiline-comment-style": Array [
      "off",
      "starred-block",
    ],
    "new-cap": Array [
      "error",
      Object {
        "capIsNew": false,
        "capIsNewExceptions": Array [
          "Immutable.Map",
          "Immutable.Set",
          "Immutable.List",
        ],
        "newIsCap": true,
        "newIsCapExceptions": Array [],
      },
    ],
    "newline-after-var": "off",
    "newline-before-return": "off",
    "no-alert": "warn",
    "no-array-constructor": "error",
    "no-async-promise-executor": "error",
    "no-await-in-loop": "error",
    "no-bitwise": "error",
    "no-buffer-constructor": "error",
    "no-caller": "error",
    "no-case-declarations": "error",
    "no-catch-shadow": "off",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": Array [
      "error",
      "always",
    ],
    "no-console": "error",
    "no-const-assign": "error",
    "no-constant-condition": "warn",
    "no-constructor-return": "error",
    "no-continue": "off",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "no-div-regex": "off",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": Array [
      "error",
      Object {
        "allowElseIf": false,
      },
    ],
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-function": Array [
      "error",
      Object {
        "allow": Array [
          "arrowFunctions",
          "functions",
          "methods",
        ],
      },
    ],
    "no-empty-pattern": "error",
    "no-eq-null": "off",
    "no-eval": "error",
    "no-ex-assign": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-boolean-cast": "error",
    "no-extra-label": "error",
    "no-extra-parens": "off",
    "no-extra-semi": "error",
    "no-fallthrough": "error",
    "no-func-assign": "error",
    "no-global-assign": Array [
      "error",
      Object {
        "exceptions": Array [],
      },
    ],
    "no-implicit-coercion": Array [
      "off",
      Object {
        "allow": Array [],
        "boolean": false,
        "number": true,
        "string": true,
      },
    ],
    "no-implicit-globals": "off",
    "no-implied-eval": "error",
    "no-import-assign": "error",
    "no-inline-comments": "off",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-invalid-this": "off",
    "no-irregular-whitespace": "error",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": Array [
      "error",
      Object {
        "allowLoop": false,
        "allowSwitch": false,
      },
    ],
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "error",
    "no-magic-numbers": Array [
      "off",
      Object {
        "detectObjects": false,
        "enforceConst": true,
        "ignore": Array [],
        "ignoreArrayIndexes": true,
      },
    ],
    "no-misleading-character-class": "error",
    "no-mixed-requires": Array [
      "off",
      false,
    ],
    "no-multi-assign": Array [
      "error",
    ],
    "no-multi-str": "error",
    "no-native-reassign": "off",
    "no-negated-condition": "off",
    "no-nested-ternary": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-require": "error",
    "no-new-symbol": "error",
    "no-new-wrappers": "error",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "error",
    "no-param-reassign": Array [
      "error",
      Object {
        "ignorePropertyModificationsFor": Array [
          "acc",
          "accumulator",
          "e",
          "ctx",
          "req",
          "request",
          "res",
          "response",
          "$scope",
          "staticContext",
        ],
        "props": true,
      },
    ],
    "no-path-concat": "error",
    "no-plusplus": "error",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-proto": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-restricted-globals": Array [
      "error",
      "isFinite",
      "isNaN",
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultStatus",
      "defaultstatus",
      "event",
      "external",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "self",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top",
    ],
    "no-restricted-imports": "off",
    "no-restricted-modules": "off",
    "no-restricted-properties": Array [
      "error",
      Object {
        "message": "arguments.callee is deprecated",
        "object": "arguments",
        "property": "callee",
      },
      Object {
        "message": "Please use Number.isFinite instead",
        "object": "global",
        "property": "isFinite",
      },
      Object {
        "message": "Please use Number.isFinite instead",
        "object": "self",
        "property": "isFinite",
      },
      Object {
        "message": "Please use Number.isFinite instead",
        "object": "window",
        "property": "isFinite",
      },
      Object {
        "message": "Please use Number.isNaN instead",
        "object": "global",
        "property": "isNaN",
      },
      Object {
        "message": "Please use Number.isNaN instead",
        "object": "self",
        "property": "isNaN",
      },
      Object {
        "message": "Please use Number.isNaN instead",
        "object": "window",
        "property": "isNaN",
      },
      Object {
        "message": "Please use Object.defineProperty instead.",
        "property": "__defineGetter__",
      },
      Object {
        "message": "Please use Object.defineProperty instead.",
        "property": "__defineSetter__",
      },
      Object {
        "message": "Use the exponentiation operator (**) instead.",
        "object": "Math",
        "property": "pow",
      },
    ],
    "no-restricted-syntax": Array [
      "error",
      Object {
        "message": "for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array.",
        "selector": "ForInStatement",
      },
      Object {
        "message": "Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand.",
        "selector": "LabeledStatement",
      },
      Object {
        "message": "\`with\` is disallowed in strict mode because it makes code impossible to predict and optimize.",
        "selector": "WithStatement",
      },
    ],
    "no-return-assign": Array [
      "error",
      "always",
    ],
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-assign": Array [
      "error",
      Object {
        "props": false,
      },
    ],
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-setter-return": "error",
    "no-shadow": "error",
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "error",
    "no-sync": "off",
    "no-template-curly-in-string": "error",
    "no-ternary": "off",
    "no-this-before-super": "error",
    "no-throw-literal": "error",
    "no-undef": "error",
    "no-undef-init": "error",
    "no-undefined": "off",
    "no-underscore-dangle": Array [
      "error",
      Object {
        "allow": Array [],
        "allowAfterSuper": false,
        "allowAfterThis": false,
        "enforceInMethodNames": true,
      },
    ],
    "no-unexpected-multiline": "error",
    "no-unmodified-loop-condition": "off",
    "no-unneeded-ternary": Array [
      "error",
      Object {
        "defaultAssignment": false,
      },
    ],
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unused-expressions": Array [
      "error",
      Object {
        "allowShortCircuit": false,
        "allowTaggedTemplates": false,
        "allowTernary": false,
      },
    ],
    "no-unused-labels": "error",
    "no-unused-vars": Array [
      "error",
      Object {
        "args": "after-used",
        "ignoreRestSiblings": true,
        "vars": "all",
      },
    ],
    "no-use-before-define": Array [
      "error",
      Object {
        "classes": true,
        "functions": true,
        "variables": true,
      },
    ],
    "no-useless-call": "off",
    "no-useless-catch": "off",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-escape": "error",
    "no-useless-rename": "error",
    "no-useless-return": "error",
    "no-var": "error",
    "no-void": "error",
    "no-warning-comments": Array [
      "off",
      Object {
        "location": "start",
        "terms": Array [
          "todo",
          "fixme",
          "xxx",
        ],
      },
    ],
    "no-with": "error",
    "object-shorthand": Array [
      "error",
      "always",
      Object {
        "avoidQuotes": true,
        "ignoreConstructors": false,
      },
    ],
    "one-var": Array [
      "error",
      "never",
    ],
    "operator-assignment": Array [
      "error",
      "always",
    ],
    "padding-line-between-statements": "off",
    "prefer-const": Array [
      "error",
      Object {
        "destructuring": "any",
        "ignoreReadBeforeAssign": true,
      },
    ],
    "prefer-destructuring": Array [
      "error",
      Object {
        "AssignmentExpression": Object {
          "array": true,
          "object": true,
        },
        "VariableDeclarator": Object {
          "array": false,
          "object": true,
        },
      },
      Object {
        "enforceForRenamedProperties": false,
      },
    ],
    "prefer-exponentiation-operator": "error",
    "prefer-named-capture-group": "off",
    "prefer-numeric-literals": "error",
    "prefer-object-spread": "off",
    "prefer-promise-reject-errors": Array [
      "error",
      Object {
        "allowEmptyReject": true,
      },
    ],
    "prefer-reflect": "off",
    "prefer-regex-literals": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "error",
    "prettier/prettier": "error",
    "radix": "error",
    "require-atomic-updates": "error",
    "require-await": "off",
    "require-jsdoc": "off",
    "require-unicode-regexp": "off",
    "require-yield": "error",
    "sort-imports": Array [
      "error",
      Object {
        "ignoreCase": false,
        "ignoreDeclarationSort": true,
        "ignoreMemberSort": false,
      },
    ],
    "sort-keys": Array [
      "off",
      "asc",
      Object {
        "caseSensitive": false,
        "natural": true,
      },
    ],
    "sort-vars": "off",
    "spaced-comment": Array [
      "error",
      "always",
      Object {
        "block": Object {
          "balanced": true,
          "exceptions": Array [
            "-",
            "+",
          ],
          "markers": Array [
            "=",
            "!",
          ],
        },
        "line": Object {
          "exceptions": Array [
            "-",
            "+",
          ],
          "markers": Array [
            "=",
            "!",
          ],
        },
      },
    ],
    "strict": Array [
      "error",
      "safe",
    ],
    "symbol-description": "error",
    "use-isnan": "error",
    "valid-typeof": Array [
      "error",
      Object {
        "requireStringLiterals": true,
      },
    ],
    "vars-on-top": "error",
    "yoda": "error",
  },
  "settings": Object {
    "import/cache": 60,
    "import/resolver": Object {
      "node": Object {
        "extensions": Array [
          ".mjs",
          ".js",
          ".json",
          ".jsx",
          ".ts",
          ".tsx",
        ],
      },
    },
  },
}
`;
